xml/tuning-os-tuning.xml

```diff
@@ -0,0 +1,425 @@
+      </mediaobject>
+     </figure>
+     <para>
+       Additionally, the settings can be made persistent by editing the
+       configuration files for the physical interfaces found in
+       <filename>/etc/sysconfig/network</filename>. A script can be
+       found in Appendix B that will change all interfaces to the
+       maximum ring buffer value.
+     </para>
+     <para>
+       The following settings can all be made persistent by modifying
+       <filename>/etc/sysctl.conf</filename>. They are represented as
+       arguments in a salt command to allow testing and validation
+       in your environment before making them permanent.
+     </para>
+     <para>
+       Setting the TCP low latency option disables IPV4 TCP pre-queue
+       processing and improves latency. We recommend experimenting
+       with this setting at both zero and one. In the lab testing that
+       was performed, setting the value to one provided slightly better performance:
+     </para>
+<screen>
+  salt '*' cmd.run 'sysctl -w net.ipv4.tcp_low_latency=1'
+</screen>
+     <para>
+       The TCP <option>fastopen</option> option allows the sending of data in the
+       first syn packet, resulting in a slight improvement in latency:
+     </para>
+<screen>
+salt '*' cmd.run 'sysctl -w net.ipv4.tcp_fastopen=1'
+</screen>
+     <para>
+       Ensure that the TCP stack has sufficent buffer space
+       to queue both inbound and outbound traffic:
+     </para>
+<screen>
+salt '*' cmd.run 'sysctl -w net.ipv4.tcp_rmem="10240 87380 2147483647"'
+salt '*' cmd.run 'sysctl -w net.ipv4.tcp_wmem="10240 87380 2147483647"'
+</screen>
+     <para>
+       In fast networks, TCP sequence numbers can be re-used in a very
+       short timeframe. The result is that the system thinks a packet
+       has been received out of order, resulting in a drop. TCP timestamps
+       were added to help ensure that packet sequence could be better tracked:
+     </para>
+<screen>
+salt '*' cmd.run 'sysctl -w net.ipv4.tcp_timestamps=1'
+</screen>
+     <para>
+       <literal>TCP Selective Acknowledgement</literal> is a feature that is primarily useful
+       for WAN or lower speed networks. However, disabling may have negative
+       effects in other ways:
+     </para>
+<screen>
+salt '*' cmd.run 'sysctl -w net.ipv4.tcp_sack=1'
+</screen>
+     <para>
+       Providing plenty of buffer space is a recurring theme in tuning
+       networks for high performance. The <literal>netdev_max_backlog</literal> is where
+       traffic is queued after it has been received by the NIC, but
+       before it is processed by the protocol stack (IP, TCP, etc):
+     </para>
+<screen>
+salt '*' cmd.run 'sysctl -w net.core.netdev_max_backlog=250000'
+</screen>
+     <para>
+       Other preventative measures for the system and application nodes
+       include ensuring that the maximum connection count is high enough
+       to prevent the generation of syn cookies. This is useful to set
```

Comment:
I am not sure if this should be:

[literal]syn[/literal] cookies

... or...

[literal]syn cookies[/literal]

Reply:
I think the latter.